### PR TITLE
Переклад book/08-customizing-git/sections/hooks.asc

### LINKED ISSUE #33: 08-customizing-git/sections/hooks.asc
Перекласти файл 08-customizing-git/sections/hooks.asc

### PR DESCRIPTION
**Pull request by hedrok**
fix #33

### CODE REVIEW

File: book/08-customizing-git/sections/hooks.asc [hunk 1]
```diff
@@ -1,126 +1,126 @@
 [[_git_hooks]]
-=== Git Hooks
+=== Гаки (hooks) Git
 
 (((hooks)))
-Like many other Version Control Systems, Git has a way to fire off custom scripts when certain important actions occur.
-There are two groups of these hooks: client-side and server-side.
-Client-side hooks are triggered by operations such as committing and merging, while server-side hooks run on network operations such as receiving pushed commits.
-You can use these hooks for all sorts of reasons.
+Як і багато інших Систем Керування Версіями, Git має спосіб запускати користувацькі скрипти, коли виконує якісь важливі дії.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: виконуЮться

File: book/08-customizing-git/sections/hooks.asc [hunk 2]
```diff
@@ -1,126 +1,126 @@
...
+Існує дві групи таких гаків: клієнтські та серверні.
+Клієнтські гаки викликаються операціями на кшталт збереження коміту та зливання, у той час як серверні гаки виконується під час мережевих операцій на кшталт отримання надісланих змін.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: виконуЮться

Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: один "на кшталт" можна замінити б на шось на кшталт "типу".

пардон за каламбур

File: book/08-customizing-git/sections/hooks.asc [hunk 3]
```diff
@@ -1,126 +1,126 @@
...
+==== Встановлення гаків
 
-The hooks are all stored in the `hooks` subdirectory of the Git directory.
-In most projects, that's `.git/hooks`.
-When you initialize a new repository with `git init`, Git populates the hooks directory with a bunch of example scripts, many of which are useful by themselves; but they also document the input values of each script.
-All the examples are written as shell scripts, with some Perl thrown in, but any properly named executable scripts will work fine – you can write them in Ruby or Python or what have you.
-If you want to use the bundled hook scripts, you'll have to rename them; their file names all end with `.sample`.
+Усі гаки зберігаються у піддиректорії `hooks` директорії Git.
+У більшості проектів, це `.git/hooks`.
+Під час ініціалізації нового репозиторію за допомогою `git init`, Git заповнює директорію гаків купою прикладів скриптів, багато з яких є корисним, а також документують вхідні значення кожного скрипта.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: репозиторію -> сховища

(?)

Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: є кориснимИ

File: book/08-customizing-git/sections/hooks.asc [hunk 4]
```diff
@@ -1,126 +1,126 @@
...
+Всі приклади написані як скрипти оболонки, з невеликими додатками на Perl, проте будь-який правильно названий виконанний скрипт спрацює – ви можете писати їх на Ruby чи Python чи на чому забажаєте.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: проте(,)

File: book/08-customizing-git/sections/hooks.asc [hunk 5]
```diff
@@ -1,126 +1,126 @@
...
+Після цього, він має викликатися.
+Ми розглянемо тут найголовніші назви файлів гаків.
 
-==== Client-Side Hooks
+==== Клієнтські гаки
 
-There are a lot of client-side hooks.
-This section splits them into committing-workflow hooks, e-mail-workflow scripts, and everything else.
+Існує багато клієнтських гаків.
+Це секція розділяє їх на гаки процесу роботи з комітами, процесу роботи з поштою, та все інше.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: Це секція->Ця секція

File: book/08-customizing-git/sections/hooks.asc [hunk 6]
```diff
@@ -1,126 +1,126 @@
...
 
 [NOTE]
 ====
-It's important to note that client-side hooks are *not* copied when you clone a repository.
-If your intent with these scripts is to enforce a policy, you'll probably want to do that on the server side; see the example in <<_an_example_git_enforced_policy>>.
+Важливо зауважити, що клієнтські гаки *не* копіюються, коли ви створюєте клон репозиторію.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: репозиторію

сховища

(?)

File: book/08-customizing-git/sections/hooks.asc [hunk 7]
```diff
@@ -1,126 +1,126 @@
...
 
-The first four hooks have to do with the committing process.
+Перші чотири гаки пов’язані з процесом створення комітів.
 
-The `pre-commit` hook is run first, before you even type in a commit message.
-It's used to inspect the snapshot that's about to be committed, to see if you've forgotten something, to make sure tests run, or to examine whatever you need to inspect in the code.
-Exiting non-zero from this hook aborts the commit, although you can bypass it with `git commit --no-verify`.
-You can do things like check for code style (run `lint` or something equivalent), check for trailing whitespace (the default hook does exactly this), or check for appropriate documentation on new methods.
+Спочатку виконується гак `pre-commit`, навіть перед тим, як ви набрали повідомлення коміту.
+Він використовується щоб оглянути відбиток, який ви збираєтесь зберегти в коміті, щоб побачити, чи забули ви щось, переконатись, що тести проходять, чи щоб перевірити будь-що, що вам потрібно в коді.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: будь-що, що вам потрібно в коді -> будь-що в коді.

спростити (?)

File: book/08-customizing-git/sections/hooks.asc [hunk 8]
```diff
@@ -1,126 +1,126 @@
...
+Вихід з цього гака з не нулем скасовує коміт, хоча щоб цього уникнути за допомогою `git commit --no-verify`.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: > не нулем

пропоную через дефіс  || з ненульовим значенням || не з нулем

File: book/08-customizing-git/sections/hooks.asc [hunk 9]
```diff
@@ -1,126 +1,126 @@
...
+Це дозволяє вам редагувати типове повідомлення перед тим, як автор коміту побачить його.
+Цей гак приймає декілька параметрів: шлях до файлу, який містить поточне повідомлення коміту, тип коміту та SHA-1 коміту, якщо це виправлення (amend) коміту.
+Цей гак зазвичай не є дуже корисним для нормальних комітів: швидше він є доречним для комітів, для яких типове повідомлення комітів автоматично генеруються, наприклад шаблонні повідомлення комітів, коміти злиття, коміти зварювання та коміти виправлення.
+Ви можете використовувати його разом з шаблоном коміту, щоб програмно додати інформацію.
 
-The `commit-msg` hook takes one parameter, which again is the path to a temporary file that contains the commit message written by the developer.
-If this script exits non-zero, Git aborts the commit process, so you can use it to validate your project state or commit message before allowing a commit to go through.
-In the last section of this chapter, We'll demonstrate using this hook to check that your commit message is conformant to a required pattern.
+Ufr `commit-msg` приймає один параметр -- знову шлях до тимчасового файлу, який містить повідомлення коміту, яке написав розробник.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: Ufr це гак, я поняв :)

Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: знову шлях до тимчасового файлу -> той самий шлях до тимчасового файлу

File: book/08-customizing-git/sections/hooks.asc [hunk 10]
```diff
@@ -1,126 +1,126 @@
...
+Якщо скрипт виходить не з нулем, то Git скасовує коміт, отже ви можете використати його, щоб перевіряти стан або повідомлення коміту перед тим, як дозволити йти далі.
+В останній секції цього розділу, ми продемонструємо використання цього гаку для перевірки відповідності повідомлення коміту необхідному взірцю.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc:50]: взірцю

:+1:

File: book/08-customizing-git/sections/hooks.asc [hunk 11]
```diff
@@ -1,126 +1,126 @@
...
 [[_email_hooks]]
-===== E-mail Workflow Hooks
+===== Гаки процесу роботи з поштою
 
-You can set up three client-side hooks for an e-mail-based workflow.
-They're all invoked by the `git am` command, so if you aren't using that command in your workflow, you can safely skip to the next section.
-If you're taking patches over e-mail prepared by `git format-patch`, then some of these may be helpful to you.
+Ви можете налаштувати три клієнтські гаки для процесу роботи з поштою.
+Вони всі викликаються командою `git am`, отже якщо ви її не використовуєте в роботі, ви можете спокійнісінько пропустити цей підрозділ.
+Якщо ви отримуєте латки через пошту, та вони були підготовлені командою `git format-patch`, то якісь з них можуть бути корисними для вас.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: бути корисними для вас->бути для вас корисними 

reorder

File: book/08-customizing-git/sections/hooks.asc [hunk 12]
```diff
@@ -1,126 +1,126 @@
...
 
-The first hook that is run is `applypatch-msg`.
-It takes a single argument: the name of the temporary file that contains the proposed commit message.
-Git aborts the patch if this script exits non-zero.
-You can use this to make sure a commit message is properly formatted, or to normalize the message by having the script edit it in place.
+Спершу виконується гак `applypatch-msg`.
+Він приймає один аргумент: імʼя тимчасового файлу, що містить пропоноване повідомлення коміту.
+Git відкидає латку, якщо скрипт виходить не з нулем.
+Ви можете використати його щоб переконатись, що повідомлення коміту написано належним чином, чи для нормалізації повідомлення -- скрипт може одразу відредагувати його.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: його(?) щоб

File: book/08-customizing-git/sections/hooks.asc [hunk 13]
```diff
@@ -1,126 +1,126 @@
...
 
-The next hook to run when applying patches via `git am` is `pre-applypatch`.
-Somewhat confusingly, it is run _after_ the patch is applied but before a commit is made, so you can use it to inspect the snapshot before making the commit.
-You can run tests or otherwise inspect the working tree with this script.
-If something is missing or the tests don't pass, exiting non-zero aborts the `git am` script without committing the patch.
+Наступний гак -- `pre-applypatch` --  викликається під час застосування латок командою `git am`
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: командою `git am`(.)

File: book/08-customizing-git/sections/hooks.asc [hunk 14]
```diff
@@ -1,126 +1,126 @@
...
+Це може збити з пантелику, проте він виконується _після_ того, як латку застосовано, проте перед тим, як створюється коміт. Отже його можна використати для перевірки відбитку перед додавання коміту.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: Отже(,)

Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: додаванняМ

File: book/08-customizing-git/sections/hooks.asc [hunk 15]
```diff
@@ -1,126 +1,126 @@
...
+Ви можете виконати тести чи іншим чином перевірити робоче дерево в цьому скрипті.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: тести(,) чи

File: book/08-customizing-git/sections/hooks.asc [hunk 16]
```diff
@@ -1,126 +1,126 @@
...
+Якщо чогось бракує, чи тести не пройшли, вихід з не нулем припиняє скрипт `git am` без створення коміту з латкою.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: > з не нулем

див вище

File: book/08-customizing-git/sections/hooks.asc (hunk 2 above)
Comment (hedrok) [inline book/08-customizing-git/sections/hooks.asc]: Замінив, я зазвичай просто якщо в оригіналі повторення, то й в перекладі роблю повторення.

File: book/08-customizing-git/sections/hooks.asc (hunk 8 above)
Comment (hedrok) [inline book/08-customizing-git/sections/hooks.asc]: Сподіваюсь ці варіанти для тебе еквівалентні - я вибрав останній, адже він коротший.
Я не проти інших - якщо хочеш, виправлю на перший чи другий.

**Comment by hedrok** [comment]
Ці виправив.

File: book/08-customizing-git/sections/hooks.asc [hunk 17]
```diff
@@ -1,126 +1,126 @@
...
+Ви можете використати його, щоб переконатися, що повідомлення коміту написано належним чином, чи для нормалізації повідомлення -- скрипт може одразу відредагувати його.
 
-The next hook to run when applying patches via `git am` is `pre-applypatch`.
-Somewhat confusingly, it is run _after_ the patch is applied but before a commit is made, so you can use it to inspect the snapshot before making the commit.
-You can run tests or otherwise inspect the working tree with this script.
-If something is missing or the tests don't pass, exiting non-zero aborts the `git am` script without committing the patch.
+Наступний гак -- `pre-applypatch` --  викликається під час застосування латок командою `git am`.
+Це може збити з пантелику, проте він виконується _після_ того, як латку застосовано, проте перед тим, як створюється коміт. Отже, його можна використати для перевірки відбитку перед додаванням коміту.
+Ви можете виконати тести, чи іншим чином перевірити робоче дерево в цьому скрипті.
+Якщо чогось бракує, чи тести не пройшли, вихід з не нульовим значенням припиняє скрипт `git am` без створення коміту з латкою.
```
Comment (tbudurovych) [inline book/08-customizing-git/sections/hooks.asc]: > не нульовим 

разом
